src/services/spotify.service.ts

```diff
@@ -0,0 +1,78 @@
+
+const BASE_URL = 'https://api.spotify.com/v1';
+
+export class SpotifyService {
+  private _accessToken: string | null = null;
```

Comment:
The access token is only ever used for making HTTP requests. The point of this service is to make pretty much all of the api calls, so storing it here will mean it gets passed around a lot less